### PR TITLE
Update babylon.scene.ts

### PR DESCRIPTION
**Pull request by dad72**
One can choose the skeleton where you want to search the bone by name or by id

### CODE REVIEW

**Comment by RaananW** [comment]
Hi David,

I am trying to understand what the difference is between the implemented method and the one you PRed here.
There are also Travis errors to fix (no need for a break after the return method), but would be great to know first what you actually did :smile:

**Comment by dad72** [comment]
I add the ability to choose the skeleton to research of a bone.
before, it was looking at all the available skeleton on scene.
Now we can choose a skeleton among those on scene

**Comment by dad72** [comment]
For the 2 first error, I corrected.

I think to find a bone by name or id, several character on stage for example has the same bone name. Seeking a bone of a skeleton, you get the skeleton that we want.

File: src/babylon.scene.ts [hunk 1]
```diff
@@ -1153,37 +1153,31 @@
...
         * get a bone using its id
         * @param {string} the bone's name
+        * @param {Skeleton} the skeleton 
         * @return {BABYLON.Bone|null} the bone or null if not found
         */
-        public getBoneByName(name: string): Bone {
-            for (var skeletonIndex = 0; skeletonIndex < this.skeletons.length; skeletonIndex++) {
-                var skeleton = this.skeletons[skeletonIndex];
-                for (var boneIndex = 0; boneIndex < skeleton.bones.length; boneIndex++) {
+        public getBoneByName(name: string, skeleton: Skeleton): Bone {
```
Comment (RaananW) [inline src/babylon.scene.ts]: This is a breaking change. I can't say if this feature is needed or not (I guess it is needed if you have two bones with the same name in different skeletons), but the skeleton variable should be optional.

File: src/babylon.scene.ts [hunk 2]
```diff
@@ -1426,10 +1424,13 @@
...
                 return camera;
             }
-
-            var bone = this.getBoneByName(name);
-
-            return bone;
+            var bone = null;
+            for (var skeletonIndex = 0, cache = this.skeletons.length; skeletonIndex < cache; skeletonIndex++) {
+                var skeleton = this.skeletons[skeletonIndex];
+		bone = this.getBoneByName(name, skeleton);
```
Comment (RaananW) [inline src/babylon.scene.ts]: If a bone was found return it, otherwise it will traverse the entire skeleton index for no reason (performance reduction)

**Comment by dad72** [comment]
By imagining a scene you have more character. you can attach an object on the hand of one of the characters. seeking the bone by the name 'right hand', my method can only select the skeleton on which you want to attach the object and not on the first skeleton found on the scene.

So I think we should keep functions as before and create new ones with a different name.
Ok, I'll do that.

**Comment by dad72** [comment]
I handed over the functions as before and have added another that better suits what I wanted to do.
This allows you to:

```
var hant_left = scene.searchIndexOnSkeletonByName(skeleton, "hant_left ");
sword.attachToBone(skeleton.bones[hant_left], character);
```

**Comment by deltakosh** [comment]
This function should be on Skeleton class

**Comment by dad72** [comment]
I thought that by putting it on scene. we needed a skeleton on scene so the bone.

But yes it could also be on skeleton. I do like that.

**Comment by dad72** [comment]
I close this pull

**Pull request closed without merging**